motion/thruster_interface_auv/src/thruster_interface_auv_driver.cpp

```diff
@@ -0,0 +1,169 @@
+#include "thruster_interface_auv/thruster_interface_auv_driver.hpp"
+
+ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(
+    short i2c_bus,
+    int pico_i2c_address,
+    const std::vector<short>& thruster_mapping,
+    const std::vector<short>& thruster_direction,
+    const std::vector<int>& pwm_min,
+    const std::vector<int>& pwm_max,
+    const std::vector<double>& left_coeffs,
+    const std::vector<double>& right_coeffs)
+    : i2c_bus_(i2c_bus),
+      pico_i2c_address_(pico_i2c_address),
+      thruster_mapping_(thruster_mapping),
+      thruster_direction_(thruster_direction),
+      pwm_min_(pwm_min),
+      pwm_max_(pwm_max),
+      left_coeffs_(left_coeffs),
+      right_coeffs_(right_coeffs) {
+    printf("I2C_BUS: %d\n", i2c_bus_);
+    printf("PICO_I2C_ADDRESS: %d\n", pico_i2c_address_);
+    printf("THRUSTER_MAPPING: ");
+    for (int i = 0; i < thruster_mapping_.size(); i++) {
+        printf("%d ", thruster_mapping_[i]);
+    }
+    printf("\n");
+    printf("THRUSTER_DIRECTION: ");
+    for (int i = 0; i < thruster_direction_.size(); i++) {
+        printf("%d ", thruster_direction_[i]);
+    }
+    printf("\n");
+    printf("PWM_MIN: ");
+    for (int i = 0; i < pwm_min_.size(); i++) {
+        printf("%d ", pwm_min_[i]);
+    }
+    printf("\n");
+    printf("PWM_MAX: ");
+    for (int i = 0; i < pwm_max_.size(); i++) {
+        printf("%d ", pwm_max_[i]);
+    }
+    printf("\n");
+    printf("LEFT_COEFFS: ");
+    for (int i = 0; i < left_coeffs_.size(); i++) {
+        printf("%f ", left_coeffs_[i]);
+    }
+    printf("\n");
+    printf("RIGHT_COEFFS: ");
+    for (int i = 0; i < right_coeffs_.size(); i++) {
+        printf("%f ", right_coeffs_[i]);
+    }
```

Comment:
Are these intended to be kept? If so, I recommend making a free-function helper:

```C++
template <typename T>
void print_vector(const std::string& name, const std::vector<T>& vec) {
    std::cout << name << ": ";
    for (const auto& elem : vec) {
        std::cout << elem << " ";
    }
    std::cout << std::endl;
}
```
which then allows you to simply:
```C++
print_vector("THRUSTER_MAPPING", thruster_mapping_);
```



Note, the below is purely fun C++ tech (IGNORE UNLESS U WANT TO SEE MODERN C++), but you could in theory generalize that to print any container that is "printable", either hacking it together with C++17 SFINAE, or C++20 concepts:

C++17:
```C++
template <typename T, typename = void>
struct is_printable : std::false_type {};

template <typename T>
struct is_printable<T, std::void_t<decltype(std::cout << std::declval<T>())>> : std::true_type {};

template <typename Container>
void printContainer(const std::string& name, const Container& container) {
    static_assert(is_printable<typename Container::value_type>::value, 
                  "Error: Container elements are not printable using std::ostream");

    std::cout << name << ":" << std::endl;
    for (const auto& elem : container) {
        std::cout << elem << " ";
    }
    std::cout << std::endl;
}
```
C++20:
```C++
template <typename Container>
void printContainer(const std::string& name, const Container& container) 
requires requires(const typename Container::value_type& x) { std::cout << x; } 
{
    std::cout << name << ":" << std::endl;
    for (const auto& elem : container) {
        std::cout << elem << " ";
    }
    std::cout << std::endl;
}
```

Also note: you could definitely use the new print format stuff also, but I was lazy and just used std:.cout :smile: